README.md

```diff
@@ -1,372 +1,332 @@
+    1. [Async iterators](#async-iterators)
+1. [Examples](#examples)
+1. [TypeScript](#typescript)
 
 # Installation
 
-If you're using Rollup/Webpack or similar:
-
 ```sh
 npm install idb
 ```
 
-Then in your JS:
+Then, assuming you're using a module-compatible system (like Webpack, Rollup etc):
 
 ```js
-import { openDb, deleteDb } from 'idb';
+import { openDB, deleteDB, wrap, unwrap } from 'idb';
 
-await openDb(…);
+async function doDatabaseStuff() {
+  const db = await openDB(…);
+}
 ```
 
-Or include [the script](https://github.com/jakearchibald/idb/blob/master/build/idb.js) as it is, and `idb` will exist on the global scope.
+# Changes
 
-# Examples
+This is a rewrite from 3.x. [See changes](changes.md).
 
-## Keyval Store
+# API
 
-This is very similar to `localStorage`, but async. If this is *all* you need, you may be interested in [idb-keyval](https://www.npmjs.com/package/idb-keyval), you can always upgrade to this library later.
+## `openDB`
 
-```js
-const dbPromise = openDb('keyval-store', 1, upgradeDB => {
-  upgradeDB.createObjectStore('keyval');
-});
+This method opens a database, and returns a promise for an enhanced [`IDBDatabase`](https://w3c.github.io/IndexedDB/#database-interface).
 
-const idbKeyval = {
-  async get(key) {
-    const db = await dbPromise;
-    return db.transaction('keyval').objectStore('keyval').get(key);
-  },
-  async set(key, val) {
-    const db = await dbPromise;
-    const tx = db.transaction('keyval', 'readwrite');
-    tx.objectStore('keyval').put(val, key);
-    return tx.complete;
-  },
-  async delete(key) {
-    const db = await dbPromise;
-    const tx = db.transaction('keyval', 'readwrite');
-    tx.objectStore('keyval').delete(key);
-    return tx.complete;
-  },
-  async clear() {
-    const db = await dbPromise;
-    const tx = db.transaction('keyval', 'readwrite');
-    tx.objectStore('keyval').clear();
-    return tx.complete;
+```js
+const db = await openDB(name, version, {
+  upgrade(db, oldVersion, newVersion, transaction) {
+    // …
   },
-  async keys() {
-    const db = await dbPromise;
-    return db.transaction('keyval').objectStore('keyval').getAllKeys(key);
+  blocked() {
+    // …
   },
-};
+  blocking() {
+    // …
+  }
+});
 ```
 
-### Usage
+* `name`: Name of the database.
+* `version`: Schema version.
+* `upgrade` (optional): Called if this version of the database has never been opened before. Use it to specify the schema for the database. This is similar to the `upgradeneeded` event in plain IndexedDB.
+    * `db`: An enhanced `IDBDatabase`.
+    * `oldVersion`: Last version of the database opened by the user.
+    * `newVersion`: Whatever new version you provided.
+    * `transaction`: The transaction for this upgrade. This is useful if you need to get data from other stores as part of a migration.
+* `blocked` (optional): Called if there are older versions of the database open on the origin, so this version cannot open.
```

Comment:
I'll add some links to MDN, but I want to avoid documenting IDB in this readme.

I'm working on a "How to use IDB" series of doc which explain how IDB works, using this library. Once that's done, I'll link to it from this readme. That way the reader can decide "I know IDB, I just want to know about this library" vs "Explain everything".